### ISSUE TITLE
angle compile error

### ISSUE THREAD

**Issue by inpeechreee** [shinchiro/mpv-winbuild-cmake#44]
Not sure if this is an upstream issue or not, my apologies. I was able to last build successfully on Friday Dec 1, 2017. The below output is from a fresh git clone to rule out any cruft/corruption/etc from rebuilding.

**Performing make-all step for 'angle'**
```
FAILED: packages/angle-prefix/src/angle-stamp/angle-make-all 
cd /home/user/mpv-winbuild-cmake/build64/packages/angle-prefix/src/angle/build && /usr/bin/cmake -P /home/user/mpv-winbuild-cmake/build64/packages/angle-prefix/src/angle-stamp/angle-make-all-.cmake && /usr/bin/cmake -E touch /home/user/mpv-winbuild-cmake/build64/packages/angle-prefix/src/angle-stamp/angle-make-all
CMake Error at /home/user/mpv-winbuild-cmake/build64/packages/angle-prefix/src/angle-stamp/angle-make-all-.cmake:16 (message):
  Command failed: 2

   '/home/user/mpv-winbuild-cmake/build64/exec' 'make' '-j12' 'CXX=x86_64-w64-mingw32-g++' 'AR=x86_64-w64-mingw32-ar' 'RANLIB=x86_64-w64-mingw32-ranlib' 'BUILDTYPE=Release'

  See also

    /home/user/mpv-winbuild-cmake/build64/packages/angle-prefix/src/angle-stamp/angle-make-all-*.log


ninja: build stopped: subcommand failed.
```

**packages/angle-prefix/src/angle-stamp/angle-make-all-err.log**

```
../src/compiler/translator/SymbolTable.cpp: In function 'constexpr const sh::TType* sh::SpecificType(const sh::TType*, int)':
../src/compiler/translator/SymbolTable.cpp:263:31: error: call to non-constexpr function 'sh::TBasicType sh::TType::getBasicType() const'
     switch (type->getBasicType())
             ~~~~~~~~~~~~~~~~~~^~
../src/compiler/translator/SymbolTable.cpp: In function 'constexpr const sh::TType* sh::VectorType(const sh::TType*, int)':
../src/compiler/translator/SymbolTable.cpp:293:31: error: call to non-constexpr function 'sh::TBasicType sh::TType::getBasicType() const'
     switch (type->getBasicType())
             ~~~~~~~~~~~~~~~~~~^~
make: *** [src/translator.target.mk:265: out/Release/obj.target/translator/src/compiler/translator/SymbolTable.o] Error 1
make: *** Waiting for unfinished jobs....
```
```
Fedora 27
Linux dumpster 4.13.16-302.fc27.x86_64 #1 SMP Thu Nov 30 15:33:36 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,objc,obj-c++,fortran,ada,go,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 7.2.1 20170915 (Red Hat 7.2.1-2) (GCC) 
```

**Comment by rld243** [comment]
Same problem on Windows 10 (Ubuntu WSL)

**Comment by DeadSix27** [comment]
@inpeechreee It's 'cuz of: https://chromium.googlesource.com/angle/angle.git/+/614dd0f537d4cc1d0390e7cfa0691c3f0c698399
Probably needs to be fixed upstream. But I haven't found any complaints about it in their issue tracker... so either it's known or not.

**Comment by shinchiro** [comment]
Looks like upstream's problem. I just filed a report about it to them. Let's see if this will be fixed or not. Meanwhile, you skip compile angle by removing this line:
https://github.com/shinchiro/mpv-winbuild-cmake/blob/9fc923e15a71f43dd6c4852942b86a95e5beb874/packages/mpv.cmake#L3

**Comment by DeadSix27** [comment]
@shinchiro Or use commit ded7923b2553d825633c18cf092770a10be47a1f, works fine in the meantime.

**Comment by DeadSix27** [comment]
@shinchiro PS: You need to update the vulkan patch, they changed the log line, feel free to copy paste it from me if you think it's correctly done: https://github.com/DeadSix27/mpv-winbuild-cmake/commit/4ebba2d1509c5add66951a5f0617291b234eb66b

**Comment by shinchiro** [comment]
Fixed in upstream google/angle@030017a
